sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -954,6 +954,15 @@ class Analyzer(
     try {
       expr transformUp {
         case GetColumnByOrdinal(ordinal, _) => plan.output(ordinal)
+        case u @ UnresolvedAttribute(nameParts)
```

Comment:
This change looks suspicious. Doesn't `ResolveMissingReferences` resolve `grouping_id` used in `order by`?